k8s/docs/change-control-process.md

```diff
@@ -0,0 +1,106 @@
+### Spack Infrastructure Change Control Process
+
+This document details the change control process for the spack infrastructure
+repository and explains the parts thereof that are automated by the various
+software components installed on the kubernetes cluster.
+
+#### 0 - Development and Iteration
+
+Although this change control process is centered around a single git repository,
+the contents are not "developed" in the same way that a code base would be.  We
+try to keep the process similar where possible for the sake of familiarity, but
+deviate where necessary.
+
+For example, there is no complete analog for local development and testing; only
+incomplete approaches with various tradeoffs.
+
+ - You could stand up your own local kubernetes cluster using something like
+   [minikube](https://minikube.sigs.k8s.io/docs/) and iterate on that, but there
+   are a number of facilities that are hard or impossible to reproduce locally,
+   such as DNS and certificate management.  Also, many of the file contents in
+   this repo contain portions that are specific to the production cluster, and
+   would likely not work for a local development cluster without modification.
+
+ - You could deploy resources that you are iterating on directly to the
```

Comment:
I've had surprisingly an easy time with this! I think for new folks it would be helpful to have a guide akin to the manual steps  you walked @alecbcs and I though. It's important to have a frame of mind for how to figure these things out, or just strategies for doing that, and for knowing to do a step, check the outcome, and debug if something is wrong.